File: baekjoon/2581/2581.py
'''
소수 python 풀이
https://doctcoder.tistory.com/entry/%EC%86%8C%EC%88%98-%ED%92%80%EC%9D%B4
'''
arr=[True]*10000
m=int(10000**0.5)
for i in range(2,m+1):
	if arr[i]==True:
		for j in range(i+i,10000,i):
			arr[j]=False
narr=[i for i in range(2,10000) if arr[i]==True]

narr.reverse()
m=int(input())
n=int(input())
sum=0
min=100000
for i in narr:
	if i<=n: 
		if i>=m:
			sum+=i
	if i>=m: min=i
if sum==0: print(-1)
else:
	print(sum)
	print(min)

File: baekjoon/2581/2581.cpp
/*
소수 C++ 풀이
https://doctcoder.tistory.com/entry/%EC%86%8C%EC%88%98-%ED%92%80%EC%9D%B4
*/
#include <iostream>
#include <cmath>

using namespace std;
int main()
{
  int m,n;
  scanf("%d",&m);
  scanf("%d",&n);
  int nums[10001]={0,};
  for(int j=2;j<10001;j++) nums[j]=j;
  for(int k=2;k<10001;k++)  // 에라토스테네스의 체
  {
    if(nums[k]==0) continue;
    for(int l=2*k;l<10001;l+=k) nums[l]=0;
  }
  int sum=0;
  int small=0;
  for(int h=n;h>=m;h--) // 최소의 소수와 소수들의 합을 찾음
  {
    if(nums[h]!=0)
    {
      sum+=nums[h];
      small=nums[h];
    }
  }
  if(sum==0) printf("%d",-1);
  else
  {
    printf("%d\n",sum);
    printf("%d",small);
  }
}
